lib/screens/home/screen.dart

```diff
@@ -1,54 +1,99 @@
-import 'package:concordium_wallet/screens/terms_and_conditions/refresh_screen.dart';
-import 'package:concordium_wallet/state.dart';
+import 'package:concordium_wallet/screens/terms_and_conditions/screen.dart';
+import 'package:concordium_wallet/services/wallet_proxy/service.dart';
+import 'package:concordium_wallet/state/network.dart';
+import 'package:concordium_wallet/state/services.dart';
+import 'package:concordium_wallet/state/terms_and_conditions.dart';
 import 'package:flutter/material.dart';
 import 'package:provider/provider.dart';
 
-class HomeScreen extends StatelessWidget {
+class HomeScreen extends StatefulWidget {
   const HomeScreen({super.key});
 
   @override
-  Widget build(BuildContext context) {
-    final tacLastCheckedAt = context.select((AppState state) => state.termsAndConditionsLastVerifiedAt);
-    //print('tacLastCheckedAt: $tacLastCheckedAt');
+  State<HomeScreen> createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> {
+  @override
+  void initState() {
+    super.initState();
 
-    // Temporary...
-    final state = context.watch<AppState>();
-    var lastAcceptedVersion = state.sharedPreferences.termsAndConditionsAcceptedVersion;
-    // print('lastAcceptedVersion: $lastAcceptedVersion');
+    // Fetch currently valid T&C version unconditionally when initializing the widget.
+    final tacAcceptance = context.read<TermsAndConditionAcceptance>();
+    final network = context.read<ActiveNetwork>();
+    final services = context.read<ServiceRepository>().networkServices[network.active]!;
+    _updateValidTac(services.walletProxy, tacAcceptance);
```

Comment:
Tried it out. It works but doesn't make any difference in this impl because there isn't any way to refetch the valid version. Can we revisit it in https://github.com/Concordium/concordium-wallet/pull/10 where I think it actually would make a difference?

Reply:
Solved in person